tests/components/lcn/test_climate.py

```diff
@@ -0,0 +1,289 @@
+"""Test for the LCN climate platform."""
+
+from unittest.mock import patch
+
+from pypck.inputs import ModStatusVar, Unknown
+from pypck.lcn_addr import LcnAddr
+from pypck.lcn_defs import Var, VarUnit, VarValue
+
+# pylint: disable=hass-component-root-import
+from homeassistant.components.climate import DOMAIN as DOMAIN_CLIMATE
+from homeassistant.components.climate.const import (
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_TEMPERATURE,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.components.lcn.helpers import get_device_connection
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_SUPPORTED_FEATURES,
+    ATTR_TEMPERATURE,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from .conftest import MockModuleConnection
+
+
+async def test_setup_lcn_climate(hass: HomeAssistant, entry, lcn_connection) -> None:
+    """Test the setup of climate."""
+    state = hass.states.get("climate.climate1")
+    assert state is not None
+    assert state.state == "heat"
+
+
+async def test_entity_state(hass: HomeAssistant, lcn_connection) -> None:
+    """Test state of entity."""
+    state = hass.states.get("climate.climate1")
+    assert state
+    assert (
+        state.attributes[ATTR_SUPPORTED_FEATURES]
+        == ClimateEntityFeature.TARGET_TEMPERATURE
+        | ClimateEntityFeature.TURN_ON
+        | ClimateEntityFeature.TURN_OFF
+    )
+    assert set(state.attributes[ATTR_HVAC_MODES]) == {HVACMode.HEAT, HVACMode.OFF}
+    assert state.attributes[ATTR_MIN_TEMP] == 0.0
+    assert state.attributes[ATTR_MAX_TEMP] == 40.0
+    assert state.attributes[ATTR_CURRENT_TEMPERATURE] is None
+    assert state.attributes[ATTR_TEMPERATURE] is None
+
+
+async def test_entity_attributes(
+    hass: HomeAssistant, entity_registry: er.EntityRegistry, entry, lcn_connection
+) -> None:
+    """Test the attributes of an entity."""
+    entity = entity_registry.async_get("climate.climate1")
+
+    assert entity
+    assert entity.unique_id == f"{entry.entry_id}-m000007-var1.r1varsetpoint"
+    assert entity.original_name == "Climate1"
```

Comment:
Consider using the `snapshot_platform` helper as it will generate `.ambr` files which fixate the properties you are testing here. It's awesome for big data structures. After you changed the code, run `pytest ./tests/components/lcn --snapshot-update`